### PR TITLE
feat: rollouts dashboard-install manifests

### PR DESCRIPTION
**Pull request by perenesenko**
Signed-off-by: Andrii Perenesenko <andrii.perenesenko@gmail.com>
Resolve: #1184 

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

**Comment by perenesenko** [comment]
PR for the task https://github.com/argoproj/argo-rollouts/issues/1184

File: manifests/dashboard-install/dashboard-clusterrolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    app.kubernetes.io/component: rollouts-dashboard
+    app.kubernetes.io/name: rollouts-dashboard-clusterrolebinding
+    app.kubernetes.io/part-of: argo-rollouts
+  name: argo-rollouts-dashboard-clusterrolebinding
```
Comment (huikang) [inline manifests/dashboard-install/dashboard-clusterrolebinding.yaml:8]: How about removing the suffix, which seems redundant. There is an issue to track this: https://github.com/argoproj/argo-rollouts/issues/1231

File: manifests/dashboard-install/dashboard-clusterrolebinding.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: argo-rollouts-dashboard-clusterrole
```
Comment (huikang) [inline manifests/dashboard-install/dashboard-clusterrolebinding.yaml:12]: ditto

**Review by huikang — COMMENTED** [review]
Just a minor regarding the names of clusterrole, clusterolebinding to make them consistent as service, deployment, sa.

File: manifests/dashboard-install/dashboard-clusterrolebinding.yaml (hunk 1 above)
Comment (perenesenko) [reply manifests/dashboard-install/dashboard-clusterrolebinding.yaml:8]: Do you mean `app.kubernetes.io/name` label should be `rollouts-dashboard`?
Or you're talking about the **name** field?

Comment (huikang) [reply manifests/dashboard-install/dashboard-clusterrolebinding.yaml:8]: The `name` field can be `argo-rollouts-dashboard`. I think this is applicable to the label too. Thanks.

Comment (harikrongali) [reply manifests/dashboard-install/dashboard-clusterrolebinding.yaml:8]: what @huikang meant is to keep as  argo-rollouts-dashboard and remove clusterrolebinding

Comment (perenesenko) [reply manifests/dashboard-install/dashboard-clusterrolebinding.yaml]: done

File: manifests/dashboard-install/dashboard-clusterrolebinding.yaml (hunk 2 above)
Comment (perenesenko) [reply manifests/dashboard-install/dashboard-clusterrolebinding.yaml]: done

**Review by huikang — COMMENTED** [review]
The manifest looks good.

One qq: is the `manifests/dashboard-install.yaml` generated from the files under `./manifests/dashboard-install/`? If so, shall we extend https://github.com/argoproj/argo-rollouts/blob/d9d12374309ec97f510b98d5fba6806396972d87/Makefile#L249

**Comment by perenesenko** [comment]
@huikang That's true, we have to extend makefile.
As I know we're building composed YAML manually. During build we just validating it's correct

**Comment by huikang** [comment]
LGTM

File: manifests/dashboard-install/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+kind: Kustomization
+
+resources:
+- dashboard-clusterrolebinding.yaml
+- dashboard-clusterrole.yaml
+- dashboard-deployment.yaml
+- dashboard-service.yaml
+- dashboard-serviceaccount.yaml
+images:
+- name: quay.io/argoproj/argo-rollouts-dashboard
```
Comment (huikang) [inline manifests/dashboard-install/kustomization.yaml:11]: Just curious if the image is available as it seems not exist at https://quay.io/organization/argoproj

Comment (perenesenko) [reply manifests/dashboard-install/kustomization.yaml]: my mistake, thanks. Updated to the correct one `quay.io/argoproj/kubectl-argo-rollouts`

**Comment by huikang** [comment]
LGTM

**Comment by jessesuen** [comment]
Also, verify codegen is failing because you need to run `make codegen`

File: manifests/dashboard-install/dashboard-clusterrole.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      - experiments
+      - experiments/finalizers
+    verbs:
+      - create
+      - get
+      - list
+      - watch
+      - update
+      - patch
+      - delete
```
[the remark this answers is not in the document]
Comment (perenesenko) [reply manifests/dashboard-install/dashboard-clusterrole.yaml]: updated

**Comment by jtai-omniex** [comment]
when will this be included in a release?

**Pull request merged**